ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/AdminChangeLogLevel.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.jsonrpc.internal.methods;
+
+import tech.pegasys.pantheon.ethereum.jsonrpc.RpcMethod;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.JsonRpcRequest;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.exception.InvalidJsonRpcParameters;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.parameters.JsonRpcParameter;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcError;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcErrorResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcSuccessResponse;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.core.config.Configurator;
+
+public class AdminChangeLogLevel implements JsonRpcMethod {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private final JsonRpcParameter parameters;
+
+  public AdminChangeLogLevel(final JsonRpcParameter parameters) {
+    this.parameters = parameters;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ADMIN_CHANGE_LOG_LEVEL.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequest request) {
+    try {
+      final Level logLevel = parameters.required(request.getParams(), 0, Level.class);
+      LOG.debug(String.format("Setting logging level to %s", logLevel.name()));
```

Comment:
HMM... I think admin alteration of logging level should always be logged, regardless.  At debug we won't see info->warn->fatal->info.  This looks a bit counterintuitive but I think should work.

```suggestion
      LOG.log(Level.OFF, String.format("Setting logging level to %s", logLevel.name()));
```

I don't think there's hooks for preventing logging of the `OFF` level, but if there are then it should be done as `LOG.fatal` level.

Reply:
If we do not want to depend on the logger we can maybe use "System.out.println" for this message?

Reply:
The target of the log may not always be the ConsoleLogger, so I think we should stick with messages on the logger.  We uses System.out in pantheon command as it is startup code and we have more assurance we are on the console at that point.

Reply:
ok you're absolutely right. I add your modification.